ui/app/src/components/datasource/DatasourceDrawer.tsx

```diff
@@ -45,17 +45,19 @@ export function DatasourceDrawer<T extends Datasource>(props: DatasourceDrawerPr
     <Drawer isOpen={isOpen} onClose={handleClickOut} data-testid="datasource-editor">
       <ErrorBoundary FallbackComponent={ErrorAlert}>
         <PluginRegistry pluginLoader={bundledPluginLoader}>
-          {isOpen && (
-            <DatasourceEditorForm
-              initialDatasourceDefinition={{ name: datasource.metadata.name, spec: datasource.spec }}
-              initialAction={action}
-              isDraft={false}
-              isReadonly={isReadonly}
-              onSave={handleSave}
-              onClose={onClose}
-              onDelete={onDelete ? () => setDeleteDatasourceDialogStateOpened(true) : undefined}
-            />
-          )}
+          <ValidationProvider>
```

Comment:
I'm wondering if performance wise we shouldn't split ValidationProvider into 3 parts, 1 for each type of resource to validate. Or at least have one for datasource & for variable, since they are the only two cases sfully independent from dashboard. Additionally maybe we could have dedicated providers just for those 2 & ValidationProvider would wrap these + panel stuff, so that you keep a single & convenient component in HelperDashboardView? IDK you'll know better than me if this would be worth it or not :D

Reply:
I think it's fine to not split